### ISSUE TITLE
Faster Search Module only find exact matches when a filter is specified

### ISSUE THREAD

**Issue by sofl88** [freescout-helpdesk/freescout#3282]
The module uses the meilisearch filter option which only matches on exact terms. While the original FS search uses SQL's Like, it is possible to search inside contents easily which is necessary for text based fields like subject and body imo.

Example:

Filter choosen `body`; search term `hello`

Faster Search Module: no results
Original FS Search: 1 match - `hello this is...`

There is a option introduced in version 1.3.0 called `attributesToSearchOn`  (https://github.com/meilisearch/product/discussions/107#discussioncomment-6607245)

This option and the term, encapsulated in double quotes as query (q), usually gives me the expected results. I see the version requirenment might be an issue but at the current state, text searches based on specific fields seem tough to me.

Thanks in advance

**Comment by freescout-helpdesk** [comment]
So how does your PHP code look like?

**Comment by sofl88** [comment]
I'm sorry I don't know understand what you mean with my code. I did not edit any code of the module (https://freescout.net/module/faster-search/). We bought this module a while back and I noted some difference in the way it worked before. For testing purpose I've done some curl requests like

```
curl \
  -X POST 'http://localhost:7700/indexes/freescout/search' \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer aSampleMasterKey' \
  --data-binary '{ "filter": "subj = \"SOME VALUE\"" }'
```
this call lead me to insight that the filter option is similar to the SQL equal (=) and not the SQL `LIKE` - like it is the case in the original freescout search https://github.com/freescout-helpdesk/freescout/blob/561f1deb077f72d3e712aba8d8d8e975630106d9/app/Conversation.php#L2207

I see this as a weakness because I'm not able to search for emails that have a specific word in the subject (and only there) - which was possible without the module.

The module holds a filter array which is later passed to a curl request like above

```
            if (!empty($filters['subject'])) {
                $filter[] = 'subj = "'.self::prepareStringFilter($filters['subject']).'"';
            }
            if (!empty($filters['body'])) {
                $filter[] = 'body = "'.self::prepareStringFilter($filters['body']).'"';
            }
```
Now after some research I read about the recently added option `attributesToSearchOn`. The curl request would look like this

```
curl \
  -X POST 'http://localhost:7700/indexes/freescout/search' \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer aSampleMasterKey' \
  --data-binary '{ "q": "\"SOME VALUE\"", "attributesToSearchOn": ["subj"] }'
```

Add another array to the module for text values like subject and body, might be a solution if the used miliesearch version is 1.3.0+

```
            if (!empty($filters['subject'])) {
                $searchBy[] = 'subj';
                $q = '"'.self::prepareStringFilter($filters['subject']).'"'; // new
            }
            if (!empty($filters['type'])) {
                $filter[] = "type = ".(int)$filters['type']; // keep
            }
            ....
                'q' => $q,
                'page' => (int)$page,
                'hitsPerPage' => Conversation::DEFAULT_LIST_SIZE,
                'filter' => $filter,
                'attributesToSearchOn' => $searchBy
```

Things get abit more difficult if multiple text filters are selected with this solution. For this case, a multi-search might be more useful

```
curl \
  -X POST 'http://localhost:7700/multi-search' \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer aSampleMasterKey' \
  --data-binary '{
    "queries": [
      {
        "q": "test1",
	"indexUid": "freescout",
	"attributesToSearchOn": ["subj"]
      },
      {
        "q": "test2",
	"indexUid": "freescout",
	"attributesToSearchOn": ["body"]
      }
    ]
  }'
```

Thanks in advance

**Comment by freescout-helpdesk** [comment]
As of now we don't have ideas how to make the search to be able to use `attributesToSearchOn` for body and subject and use other filters at the same time. We've added a link to this thread to the module's page https://freescout.net/module/faster-search/. If someone will come up with a working solution - please let us know.

**Comment by freescout-helpdesk** [comment]
> where / how you like it to be shared?

You can send it to support@freescout.net

**Comment by freescout-helpdesk** [comment]
We've released the update as Faster Search Module v1.0.9 with some minor adjustments.

**Comment by sofl88** [comment]
@freescout-helpdesk The latest release of meilisearch (https://github.com/meilisearch/meilisearch/releases/tag/v1.10.0) comes with a new feature called federated search. With this new feature, the intersection problematic can be solved. I just want to inform you in case you want to enhance the module in the future.
